### ISSUE TITLE
false deadcode positive for main_test.go files

### ISSUE THREAD

**Issue by mastertinner** [golangci/golangci-lint#131]
`deadcode` produces false positives if there is a `main_test.go` file which is part of the `main` package. To reproduce, create the following files:

`main.go`:

```go
package main

import "fmt"

func main() {
	fmt.Println("derp")
}
```

`main_test.go`:

```go
package main

import (
	"fmt"
	"testing"
)

func TestMain(t *testing.T) {
	fmt.Println("herp")
}
```

This produces the following error message when running `golangci-lint run`:

```txt
main_test.go:8:6: `TestMain` is unused (deadcode)
func TestMain(t *testing.T) {
     ^
```

The false positive doesn't occur, if the package of the testing file is changed to `main_test`.

**Comment by jirfag** [comment]
thank you for reporting!

**Comment by jirfag** [comment]
Fixed it

**Comment by mastertinner** [comment]
Thanks!
